auparse/auditd-config.c

```diff
@@ -323,8 +327,40 @@ static int log_file_parser(auparse_state_t *au, const char *val, int line,
 	return 0;
 }
 
+
+static int eoe_timeout_parser(auparse_state_t *au, const char *val, int line,
+		struct daemon_conf *config)
+{
+	const char *ptr = val;
+	unsigned long i;
+
+	/* check that all chars are numbers */
+	for (i=0; ptr[i]; i++) {
+		if (!isdigit(ptr[i])) {
+			audit_msg(au, LOG_ERR,
+				"Value %s should only be numbers - line %d",
+				val, line);
+			return 1;
+		}
+	}
+
+	/* convert to unsigned long */
+	errno = 0;
+	i = strtoul(val, NULL, 10);
+	if (errno) {
+		audit_msg(au, LOG_ERR,
+			"Error converting string to a number (%s) - line %d",
+			strerror(errno), line);
+		return 1;
+	}
+	config->end_of_event_timeout = i;
+	return 0;
+}
+
 void free_config(struct daemon_conf *config)
 {
-	free((void*)config->log_file);
+	if (config->log_file == NULL)
```

Comment:
free() is advertised as just returning if NULL is passed so that we do not need to check if its NULL before calling free. Besides, I think you meant !=, but its totally unnecessary.

Reply:
Correct, my bad. I did mean !=.

That said, in ausearch/aureport I was seeing the error attempting to double free memory, so wanted to remove the error message. If we don't do this, then every execution will do a double free, for we allocate this structure twice, meaning free_config() is called twice.

I suppose I was just being consistent between ausearch's free_config() and auparse's. Technically, the auparse code doesn't need this 'error message protection', but this is just in case of the sequence

```
struct daemon_conf config;
au_load_config(au, &config, ..);
do stuff;
free_config(&config);
au_load_config(au, &config, ..);
do stuff
free_config(&config);
```

Reply:
yes for deleting the 'if'. Setting to NULL is fine

Reply:
Done